### DISCUSSION TITLE
RxApp.SuspensionHost.GetAppState<AppState>() has changed behaviour 0.10->11.0

### DISCUSSION THREAD

**Discussion by LosManos** [AvaloniaUI/Avalonia#13895] [Q&A] [answered]
When my 0.10 code calls
`var state = RxApp.SuspensionHost.GetAppState<AppState>();` an object was returned.
When my 11.0 code does the same call null is returned.

I guess it is not a bug - as it is so basic that many systems would crash.
But on the other hand - I don't see how to get around it.

What is the solution?

----

I did a [dirty workaround just to get the code working](https://github.com/LosManos/PurpleExplorer/commit/9170751cefc9c83a3a7cc73b372a0a5a8e5a8850?diff=split&w=1#diff-70c3fcf92aa72857516d14044a94a323d3fc5e5541512ffa24e01ec2528817b3). But then the application stopped using its settings.

I have done a more [Stack Overflow-ish rephrasing of the observation](https://stackoverflow.com/questions/77636293/avalonia-11-rxapp-suspensionhost-getappstateappstate-returns-null-while-v-0).

**Comment by maxkatz6** [comment, marked as answer]
RxApp is not controlled by the Avalonia. And it cannot change behavior with avalonia version changes. 
Most likely, ReactiveUI was also upgraded in your project, as they own this type.

**Comment by LosManos** [comment]
This might be another issue. Investigating...

When there is no file to read null is returned.
When there is a (blank) file an object is returned.
So I guess the behaviour makes sense.


# Answer below.

But what I found is that my Rider@macm2 removes the appsettings.json after the build.
Note: The build, not running/debug.

I have no build event that does so.
I even found the commit that changes the behaviour.

In MainWindow.xaml.cs:
```
    public override async void Show()
    {
        base.Show();
        var appState = Locator.Current.GetService<IAppState>() as AppState ?? throw new Exception("Unknown AppState");
        await ModalWindowHelper.ShowModalWindow<ConnectionStringWindow>(appState);
    }
```
Digging further I followed `await ModalWindowHelper.ShowModalWindow<ConnectionStringWindow>(appState);`
which calls 
```
        var mainWindow = (Application.Current.ApplicationLifetime as IClassicDesktopStyleApplicationLifetime)
            .Windows[0];
```
And at design time there is no `Window` and something in the designer decides the application state is null and removes the file.

The remedy is to not call att design time.
```
        if( Avalonia.Controls.Design.IsDesignMode == false)
            await ModalWindowHelper.ShowModalWindow<ConnectionStringWindow>(appState);
```

Phew. That took a while to figure out.

So my Question was wrong to start with.
Thanks @maxkatz6 for your time spent.
